Samples/AudioCreation/cs/AudioCreation/Scenario7_RecordPlayback.xaml.cs

```diff
@@ -0,0 +1,570 @@
+
+                isRecording = true;
+                requestStartRecording = false;
+                maxIndex = 0;
+                currentIndex = 0;
+                audioFrameCount = 0;
+                nonZeroSampleCount = 0;
+                recordingStartTime = DateTime.Now;
+                lastCapacityEncountered = maxCapacityEncountered = 0;
+            }
+
+            if (isRecording)
+            {
+                Debug.Assert(!isPlaying);
+                Debug.Assert(!requestStartPlaying);
+                Debug.Assert(!requestStartRecording);
+
+                HandleIncomingAudio();
+
+                /*
+                int currentStatusMsec = DateTime.Now.Millisecond;
+                // update status every tenth of a second
+                if (lastStatusMsec % 100 > currentStatusMsec % 100)
+                {
+                    CoreApplication.MainView.CoreWindow.Dispatcher.TryRunAsync(
+                        Windows.UI.Core.CoreDispatcherPriority.Normal,
+                        UpdateStatusWhileRecording);
+                }
+                lastStatusMsec = currentStatusMsec;
+                */
+            }
+        }
+
+        /// <summary>
+        /// Handle a frame of incoming audio from the input device.
+        /// </summary>
+        private unsafe void HandleIncomingAudio()
+        {
+            AudioFrame frame = frameOutputNode.GetFrame();
+
+            using (AudioBuffer buffer = frame.LockBuffer(AudioBufferAccessMode.Read))
+            using (IMemoryBufferReference reference = buffer.CreateReference())
+            {
+                byte* dataInBytes;
+                uint capacityInBytes;
+
+                // Get the buffer from the AudioFrame
+                ((IMemoryBufferByteAccess)reference).GetBuffer(out dataInBytes, out capacityInBytes);
+
+                if (capacityInBytes == 0)
+                {
+                    // we don't count zero-byte frames... and why do they ever happen???
+                    return;
+                }
+
+                // Must be multiple of 8 (2 channels, 4 bytes/float)
+                Debug.Assert((capacityInBytes & 0x7) == 0);
+
+                lastCapacityEncountered = capacityInBytes;
+                maxCapacityEncountered = Math.Max(maxCapacityEncountered, capacityInBytes);
+
+                uint bufferStart = 0;
+                if (maxIndex == 0)
+                {
+                    // if maxCapacityEncountered is greater than the audio graph buffer size, 
+                    // then the audio graph decided to give us a big backload of buffer content
+                    // as its first callback.  Not sure why it does this, but we don't want it,
+                    // so take only the tail of the buffer.
+                    if (maxCapacityEncountered > (graph.LatencyInSamples << 3))
+                    {
+                        bufferStart = capacityInBytes - (uint)(graph.LatencyInSamples << 3);
```

Comment:
How could maxCapacityEncountered be large from a previous processing pass?  It is reset to 0 when recording starts (in Graph_QuantumStarted, in the "if (requestStartRecording)" statement).  I assume by "processing pass" you mean "sequence of QuantumStarted events while recording is active"?